apps/cookielifetime.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Check if root
+root_check
+
+COOKIE_LIFETIME=$(whiptail --radiolist  "Configure after what time (in seconds) after every Login every Nextcloud user gets logged out in the Browser\nSelect one with the [ARROW] Keys and the [SPACE] key and confirm by pressing [ENTER]" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"1800s" "half an hour" ON \
+"7200s" "two hours" OFF \
+"43200s" "half a day" OFF \
+"172800s" "two days" OFF \
+"604800s" "one week" OFF \
+"2419200s" "four weeks" OFF \
+"Custom" "setup a custom time" OFF 3>&1 1>&2 2>&3)
+
+if [ "$COOKIE_LIFETIME" == "1800s" ]
+then 
+    occ_command config:system:set remember_login_cookie_lifetime --value="1800"
+elif [ "$COOKIE_LIFETIME" == "7200s" ]
+then 
+    occ_command config:system:set remember_login_cookie_lifetime --value="7200"
+elif [ "$COOKIE_LIFETIME" == "43200s" ]
+then 
+    occ_command config:system:set remember_login_cookie_lifetime --value="43200"
+elif [ "$COOKIE_LIFETIME" == "172800s" ]
+then 
+    occ_command config:system:set remember_login_cookie_lifetime --value="172800"
+elif [ "$COOKIE_LIFETIME" == "604800s" ]
+then 
+    occ_command config:system:set remember_login_cookie_lifetime --value="604800"
+elif [ "$COOKIE_LIFETIME" == "2419200s" ]
+then 
+    occ_command config:system:set remember_login_cookie_lifetime --value="2419200"
+elif [ "$COOKIE_LIFETIME" == "Custom" ]
+then 
+    while true
+    do
+        COOKIE_LIFETIME=$(whiptail --inputbox "Please enter the Cookie Lifetime in seconds, so e.g. 1800 for half an hour or 3600 for an hour\nIt is not recommended to set it to less than hafl an hour!" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+        COOKIE_LIFETIME=${COOKIE_LIFETIME//[!0-9]/}
+        if [ "$COOKIE_LIFETIME" -lt "1800" ]
```

Comment:
what would you say should be the lowest value here?

Reply:
The lowest value should be the default set before this PR came alive.

People will get issues with uploads on slow connections of large files otherwise.

Reply:
Maybe add in the info that the user will get forcefully logged out even if a upload is in progress?

Reply:
Make sure the user puts a **number** between 1800 and ---> infinity.

If not a number, msg_box in a while loop. Thanks!

If a value is wrong in the config file Nextcloud will freak out. ;)

Reply:
I have already done sure that it is not below 1800 seconds, which is the default value before this PR came alive. Also I already have made sure that it is always a number (the algorithm deletes everything, which isn't 0-9)

Reply:
> Maybe add in the info that the user will get forcefully logged out even if a upload is in progress?

Does this happen each time you change the value?
Or just if the time is up?

Reply:
In the while loop, add a `msg_box` which explain what's expected.

Thanks!

Reply:
I added a description here. 👍

Reply:
everything in this review should be done. 👍